aip/general/0129.md

```diff
@@ -0,0 +1,119 @@
+---
+id: 129
+state: draft
+created: 2023-06-21
+placement:
+  category: resource-design
+  order: 65
+---
+
+# Server-Modified Values and Defaults
+
+Server-modified and default values tend to cause problems for
+[state-driven clients](state-driven-clients). These clients are often unable to
+tell when their desired state matches the current state for these fields, as the
+rules by which a server may modify and return values are not public and
+repeatable. This AIP aims to clarify what behaviors can be expected from
+services and describing how they can be repeated on the client-side.
+
+## Guidance
+
+### Effective Values
+There are instances where a service will allocate, generate, or calculate a
+value if the client chooses not to specify one. For example: a client may
+create a virtual machine without specifying a static IP address for the virtual
+machine to be available on. Such a scenario is opting into dynamic IP address
+allocation.
+
+Some examples of these types of fields are ones that are:
+
+* generated (UUID)
+* allocated (dynamic IP address)
+* assigned (most recent software package version)
+
+An attribute with an effective value ***must*** be expressed as two fields in
+the API:
+* a mutable field that can be optionally set by the user and **must not** be
+modified by the service
+* an `OUTPUT_ONLY` field that records the effective value decided on by the
+service
+
+Example:
+```proto
+message VirtualMachine {
+  …
+  string ip_address = 4;
+  string effective_ip_address = 5 [
+    (google.api.field_behavior) = OUTPUT_ONLY
+  ];
+}
+```
+
+#### Naming
+
+Effective values *should* be named by prefixing `effective_` to the mutable
+field's name. Consistent naming is important for identifying standard behavior
```

Comment:
The "consistent naming" part is a rationale, so it could be moved to a separate rationale section.

That aside, as we already know, a *should* isn't enough for a client to infer standard behavior. 

If it's intended to be programmatically inferable, then it's a *must*. Otherwise I'd drop the statement that says it's used to identify standard behavior.

Reply:
Moved to rationale & changed to **must**